packages/datadog-instrumentations/src/memcached.js

```diff
@@ -17,35 +16,29 @@ addHook({ name: 'memcached', versions: ['>=2.2'] }, Memcached => {
       return command.apply(this, arguments)
     }
 
-    const callbackResource = new AsyncResource('bound-anonymous-fn')
-    const asyncResource = new AsyncResource('bound-anonymous-fn')
-
     const client = this
 
-    const wrappedQueryCompiler = asyncResource.bind(function () {
+    const wrappedQueryCompiler = function () {
       const query = queryCompiler.apply(this, arguments)
-      const callback = callbackResource.bind(query.callback)
-
-      query.callback = shimmer.wrapFunction(callback, callback => asyncResource.bind(function (err) {
-        if (err) {
-          errorCh.publish(err)
-        }
-        finishCh.publish()
-
-        return callback.apply(this, arguments)
-      }))
-      startCh.publish({ client, server, query })
 
+      const ctx = { client, server, query }
+      startCh.runStores(ctx, () => {
+        query.callback = shimmer.wrapFunction(query.callback, callback => function (err) {
+          if (err) {
+            ctx.error = err
+            errorCh.publish(ctx)
+          }
+          finishCh.publish(ctx)
```

Comment:
This is actually calling `publish` twice since the `runStores` on the next line will also call `publish`.

Reply:
@rochdev @wconti27 could either of you open a fix for that?